library/std/src/path.rs

```diff
@@ -2568,6 +2568,31 @@ impl Path {
         fs::metadata(self).map(|m| m.is_dir()).unwrap_or(false)
     }
 
+    /// Returns true if the path exists on disk and is pointing at a symbolic link file.
+    /// This method can alse be used to check whether symlink exists.
+    ///
+    /// This function will not traverse symbolic links.
+    /// In case of broken symbolic links this will also return true.
+    ///
+    /// If you cannot access the directory containing the file, e.g., because of a
+    /// permission error, this will return false.
```

Comment:
`In case of broken symbolic links`
I've copied this from a doc-comment of a method `is_dir` (and others), should I change their documentation too?

Reply:
I'm in the middle of writing a PR to change that last sentence on `is_dir`, etc (it's platform specific behaviour). I can perhaps include the other changes too?

Reply:
I think it would be better to merge links support separately, but update doc-comments is more suitable for your PR.

Reply:
No, `is_dir` follows symlinks, so in case of a symlink->symlink->symlink->dir, that's multiple symlinks. (Though I guess technically only one can be broken, since traversal stops there.) `is_symlink` however, will not traverse them, so there's only one symlink.

In addition, any of the other components of the path can be a symlink too. E.g. `/something/symlink1/something/symlink2`. `is_symlink` returns false if the first one is broken (because then the full path doesn't even exist), but true if the second one is broken. Changing the wording to 'a broken symlink' makes it a bit clearer that this is just about the one symlink that the path refers to, and that we're not talking about all the ways broken symlinks can be involved.  Maybe `In case the symlink is broken, ..` is a bit clearer though.

Reply:
Agree